tests/rptest/tests/topic_creation_test.py

```diff
@@ -151,17 +153,48 @@ def topic_autocreate_test(self):
         auto_topic_spec = self.kafka_tools.describe_topic(auto_topic)
         assert auto_topic_spec.retention_ms is None
         assert auto_topic_spec.retention_bytes is None
+        assert auto_topic_spec.cleanup_policy is not None
 
         # Create topic by hand, compare its properties to the autocreated one
         self.rpk.create_topic(manual_topic)
         manual_topic_spec = self.kafka_tools.describe_topic(auto_topic)
         assert manual_topic_spec.retention_ms == auto_topic_spec.retention_ms
         assert manual_topic_spec.retention_bytes == auto_topic_spec.retention_bytes
+        assert manual_topic_spec.cleanup_policy == auto_topic_spec.cleanup_policy
 
         # Clear name and compare the rest of the attributes
         manual_topic_spec.name = auto_topic_spec.name = None
         assert manual_topic_spec == auto_topic_spec
 
+        # compare topic configs as retrieved by rpk.
+        # describe the topics and convert the resulting dict in a set, to compute the difference
+        auto_topic_rpk_cfg = set(
+            self.rpk.describe_topic_configs(auto_topic).items())
+        manual_topic_rpk_cfg = set(
+            self.rpk.describe_topic_configs(manual_topic).items())
+
+        # retrieve the cloud storage mode and append it as an extra config, to check it. see issue/13492
+        auto_topic_rpk_cfg.add(('cloud_storage_mode',
+                                self.admin.get_partition_cloud_storage_status(
+                                    auto_topic, 0)['cloud_storage_mode']))
+        manual_topic_rpk_cfg.add(
+            ('cloud_storage_mode',
+             self.admin.get_partition_cloud_storage_status(
+                 manual_topic, 0)['cloud_storage_mode']))
+
+        self.logger.debug(f"{auto_topic=} config={auto_topic_rpk_cfg}")
+        self.logger.debug(f"{manual_topic=}, config={manual_topic_rpk_cfg}")
```

Comment:
praise: `f"{auto_topic=}` this is a cool way to format output in shorthand

Reply:
Yeah, I hadn't seen this before either. All credit to @andijcr